executor/src/witgen/jit/processor.rs

```diff
@@ -283,9 +298,25 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> Processor<'a, T, FixedEv
         loop {
             let identity = identity_queue.next();
             let updated_vars = match identity {
-                Some((identity, row_offset)) => {
-                    witgen.process_identity(can_process.clone(), identity, row_offset)
-                }
+                Some((identity, row_offset)) => match identity {
+                    Identity::Polynomial(PolynomialIdentity { id, expression, .. }) => {
+                        witgen.process_polynomial_identity(*id, expression, row_offset)
+                    }
+                    Identity::BusSend(BusSend {
+                        bus_id: _,
+                        identity_id,
+                        selected_payload,
+                    }) => witgen.process_call(
+                        can_process.clone(),
+                        *identity_id,
+                        &selected_payload.selector,
+                        selected_payload.expressions.len(),
```

Comment:
Why don't we just pass a reference to the `BusSend` struct? Similarly I think that would be good for the Polynomial case.

Reply:
If we pass a whole BusSend, it gives the impression that it processes the arguments instead of working with the variables.